File: 0066-plus-one/0066-plus-one.go
func plusOne(digits []int) []int {
    n := len(digits)
    for i := n - 1 ; i >= 0 ; i-- {

        if digits[i] + 1 != 10 {
            digits[i] += 1;
            return digits
        }
        digits[i] = 0

       
    }

    return append([]int{1} , digits...)

   
    
}

File: 0066-plus-one/0066-plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        l = len(digits) - 1
        carry = 1

        for i in range(l , -1 , -1):
            num = digits[i] + carry 

            if num == 10:
                carry = 1
                digits[i] = 0
            else:
                digits[i] = num
                carry = 0
                break
        
        if carry == 1:
            return [1] + digits

        return digits
            
        